### ISSUE TITLE
[feature request] encrypted notes vault

### ISSUE THREAD

**Issue by 4rtemis-4rrow** [Kin69/EasyNotes#15]
title says it all, if I knew kotlin I'd have implemented this myself, just an encrypted password protected notes vault

**Comment by Kin69** [comment]
Hey @4rtemis-4rrow, when opening the vault, would you prefer to enter a password each time or have it securely stored (encrypted) in the app? Additionally, should we include biometric authentication for opening?

**Comment by 4rtemis-4rrow** [comment]
for me personally, I'd prefer to enter the password every time, but that's just me, after all, I do value privacy and security more than user friendly-ness

**Comment by Kin69** [comment]
I've implemented a secure password mechanism using a vault button that prompts for a password. For enhanced security, I've utilized EncryptedSharedPreferences provided by Android. When you first enter a password, it's securely stored. Subsequently, the application uses this password to decrypt notes stored in a vault. Each time you access the vault, you must enter the password again.

Even if someone were to gain unauthorized access to the device and database, they would only see encrypted names and descriptions of the notes within the vault. The stored password in preferences is also encrypted, ensuring it remains inaccessible.

Additionally, due to vulnerabilities associated with brute force attacks, I have decided against implementing a numpad for entry.
